lib/lotus/model/adapters/sql/command.rb

```diff
@@ -55,10 +59,41 @@ def update(entity)
           # @api private
           # @since 0.1.0
           def delete
-            @collection.delete
+            _rescue_database_error do
+              @collection.delete
+            end
           end
 
           alias_method :clear, :delete
+
+          private
+
+          # Rescues Sequel::DatabaseError in yielded block
+          #
+          # @api private
+          def _rescue_database_error
+            yield
+          rescue Sequel::DatabaseError => e
+            raise _transform_database_error(e)
+          end
+
+          # Transforms error into a Lotus::Model::Error
+          #
+          # @api private
+          def _transform_database_error(error)
+            case error
+            when Sequel::CheckConstraintViolation
```

Comment:
@Moratorius I think all these cases can be handled by `Lotus::Model::Error`. So you can remove entirely this method.
```